### PR TITLE
 Adding nidigital hram read all samples unit test

### PR DESCRIPTION
**Pull request by ni-jfitzger**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).

- [x] ~~I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~~

- [x] I've added tests applicable for this pull request

### What does this Pull Request accomplish?

Adds nidigital unit test `test_fetch_history_ram_cycle_information_samples_to_read_all`

### List issues fixed by this Pull Request below, if any.

* #1273 is fixed by this PR.

### What testing has been done?

New unit test has been executed and passed.

### CODE REVIEW

**Comment by marcoskirsch** [comment]
test this please

File: src/nidigital/unit_tests/test_nidigital.py [hunk 1]
```diff
@@ -100,3 +148,105 @@ def test_fetch_history_ram_cycle_information_samples_to_read_too_much(self):
...
                 session.sites[1].fetch_history_ram_cycle_information(position=3, samples_to_read=5)
+
+    # Helper function for mocking multiple calls
+    def niDigital_FetchHistoryRAMCycleInformation_looping(self, vi, site, sample_index, pattern_index, time_set_index, vector_number, cycle_number, num_dut_cycles):  # noqa: N802
+        sample_index_int = int(sample_index.value)
+        pattern_index[0] = self.pattern_indices_looping[sample_index_int]
+        time_set_index[0] = self.time_set_indices_looping[sample_index_int]
+        vector_number[0] = self.vector_numbers_looping[sample_index_int]
+        cycle_number[0] = self.cycle_numbers_looping[sample_index_int]
+        num_dut_cycles[0] = self.num_duty_cycles_looping[sample_index_int]
```
[the remark this answers is not in the document]
Comment (ni-jfitzger) [reply src/nidigital/unit_tests/test_nidigital.py]: Done, although I had to leave the lists/arrays in niDigital_FetchHistoryRAMCyclePinData_looping unchanged.

File: src/nidigital/unit_tests/test_nidigital.py [hunk 2]
```diff
@@ -100,3 +148,105 @@ def test_fetch_history_ram_cycle_information_samples_to_read_too_much(self):
...
+        self.side_effects_helper['GetPatternPinList']['pinList'] = ','.join(pin_list)
+        with nidigital.Session('') as session:
+            history_ram_cycle_info = session.sites[1].fetch_history_ram_cycle_information(
+                position=0,
+                samples_to_read=-1)
+            assert self.patched_library.niDigital_FetchHistoryRAMCycleInformation.call_count == 7
+            assert self.patched_library.niDigital_GetPatternName.call_count == 2  # there's only one pattern, so this is a 2
+            assert self.patched_library.niDigital_GetTimeSetName.call_count == 6  # 3 time sets, so this is a 6
+            assert self.patched_library.niDigital_FetchHistoryRAMScanCycleNumber.call_count == 7
+            assert self.patched_library.niDigital_FetchHistoryRAMCyclePinData.call_count == 20  # 10 lists of PinStates
```
[the remark this answers is not in the document]
Comment (ni-jfitzger) [reply src/nidigital/unit_tests/test_nidigital.py]: Fixed.

File: src/nidigital/unit_tests/test_nidigital.py [hunk 3]
```diff
@@ -100,3 +148,105 @@ def test_fetch_history_ram_cycle_information_samples_to_read_too_much(self):
...
+            assert scan_cycle_numbers == [-1, 0, 1, -1, -1, -1, -1]
+
+            pin_names = session.get_pattern_pin_names('new_pattern')
+            assert self.patched_library.niDigital_GetPatternPinList.call_count == 2
+            assert pin_names == pin_list
+
+        expected_pin_states = [i.expected_pin_states for i in history_ram_cycle_info]
+        assert expected_pin_states == self.expected_pin_states_looping
+
+        # If test expects actual pin state to be 'X', then value returned by the returned can be anything.
```
[the remark this answers is not in the document]
Comment (ni-jfitzger) [reply src/nidigital/unit_tests/test_nidigital.py]: Fixed.

**Comment by sbethur** [comment]
test this please

**Pull request merged**